drivers/display/display_ist3931.c

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (c) 2024 Shen Xuyang <shenxuyang@shlinyuantech.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT ist_ist3931
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#define LOG_LEVEL CONFIG_DISPLAY_LOG_LEVEL
+LOG_MODULE_REGISTER(ist3931);
+
+#include "display_ist3931.h"
+
+typedef bool (*ist3931_bus_ready_fn)(const struct device *dev);
+typedef int (*ist3931_write_bus_fn)(const struct device *dev, uint8_t *buf, size_t len,
+				    bool command);
+
+union ist3931_bus {
+	struct i2c_dt_spec i2c;
+};
+
+struct ist3931_config {
+	union ist3931_bus bus;
+	ist3931_bus_ready_fn bus_ready;
```

Comment:
This is a good idea- it will make abstracting this driver easier in the future. However, it doesn't seem like you actually use this pointer (just call `ist3931_bus_ready` directly). I am ok with either implementation method (IE using a function pointer or a generically named function), but please remove the code related to the method that isn't used.

At a broader level, please make sure you're consistent with whichever method you choose to abstract the bus. IE if you choose to use generically named functions (`ist3931_bus_ready`, `ist3931_write_bus`) be consistent there, or define function pointers for the write bus and bus ready functions

Reply:
Thanks, I will fix the problem.

Reply:
This issue still needs to be fixed